### PR TITLE
Improvement to the B2B sample application

### PR DESCRIPTION
**Pull request by Achintha444**
## Purpose
-  Remove the username part from the add user modal and set email as the username.
- Remove the unrelated permissions from the permission list in the role management section

### CODE REVIEW

File: b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/getStartedSectionComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (thivi) [inline b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/getStartedSectionComponent.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```

File: b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/getStartedSectionComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                <GetStartedText />
+
+            </FlexboxGrid.Item>
+
+            <FlexboxGrid.Item colspan={ 9 }>
+
+                <Image src={ getStartedImage } alt="profile image" width={ 500 } />
+            </FlexboxGrid.Item>
+
+        </FlexboxGrid>
```
Comment (thivi) [inline b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/getStartedSectionComponent.tsx:44]: Can we remove the unnecessary lines?

File: b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/otherComponents/getStartedText.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (thivi) [inline b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/otherComponents/getStartedText.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```

File: b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/blogSection/otherComponents/latestNewsComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (thivi) [inline b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/blogSection/otherComponents/latestNewsComponent.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```

File: b2b-sample/libs/shared/ui/ui-basic-components/src/lib/components/customHtmlHeading/customHtmlHeading.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            return <h5 style={ style }>{ content }</h5>;
+            
+
+        case "h6":
+            return <h6 style={ style }>{ content }</h6>;
+            
+        default:
+            return;
+            
+    }
```
Comment (thivi) [inline b2b-sample/libs/shared/ui/ui-basic-components/src/lib/components/customHtmlHeading/customHtmlHeading.tsx:62]: Remove redundant lines

File: b2b-sample/libs/business-app/ui-assets/src/lib/data/sideNav-GuardioInsurance.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        },
+       
+        {
+            "title": "Blogs",
+            "eventKey": "4",
+            "icon": "PageIcon",
+            "type": "item"
+        }
+    ]
+}
```
Comment (thivi) [inline b2b-sample/libs/business-app/ui-assets/src/lib/data/sideNav-GuardioInsurance.json:31]: Add a new line

File: b2b-sample/libs/business-app/ui-assets/src/lib/data/sideNav-EasyMeetings.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        },
+       
+        {
+            "title": "Blogs",
+            "eventKey": "4",
+            "icon": "PageIcon",
+            "type": "item"
+        }
+    ]
+}
```
Comment (thivi) [inline b2b-sample/libs/business-app/ui-assets/src/lib/data/sideNav-EasyMeetings.json:30]: Add a new line

File: b2b-sample/config.json [hunk 8]
```diff
@@ -43,18 +43,19 @@
...
+            "SharedApplicationName": "EasyMeeting",
+            "UserStore" : "DEFAULT"
         }
     },
     "BusinessAdminAppConfig": {
         "AuthorizationConfig": {
-            "ClientId": "<CLIENT ID OF THE BUSINESS ADMIN APP>",
-            "ClientSecret": "<CLIENT SECRET OF THE BUSINESS ADMIN APP>"
+            "ClientId": "3i3NnXzEreOgeIT9pXdz5cZ52EIa",
+            "ClientSecret": "ajJMUHQgf8pQNCKUUFm4ClAKiwQa"
```
Comment (thivi) [inline b2b-sample/config.json:58]: Revert these

File: b2b-sample/config.json [hunk 9]
```diff
@@ -14,8 +14,8 @@
     },
     "BusinessAppConfig": {
         "AuthorizationConfig": {
-            "ClientId": "<CLIENT ID OF THE BUSINESS APP>",
-            "ClientSecret": "<CLIENT SECRET OF THE BUSINESS APP>"
+            "ClientId": "3IUFkGr4sjfDRXDm5z9fm34PgMwa",
+            "ClientSecret": "EI3KG6Z1PhnBqohWbt1P7qZA7Dsa"
```
Comment (thivi) [inline b2b-sample/config.json:18]: Revert these

File: b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/getStartedSectionComponent.tsx (hunk 2 above)
Comment (thivi) [inline b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/getStartedSectionComponent.tsx:44]: Can we remove the unnecessary lines?

File: b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/blogSection/otherComponents/latestNewsComponent.tsx (hunk 4 above)
Comment (thivi) [inline b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/blogSection/otherComponents/latestNewsComponent.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```

File: b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/meetingsSection/meetingsSectionComponent.tsx [hunk 10]
```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (thivi) [inline b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/meetingsSection/meetingsSectionComponent.tsx:2]: Fix the year

File: b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/phoneSection/otherComponents/phoneComponent.tsx [hunk 11]
```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (thivi) [inline b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/phoneSection/otherComponents/phoneComponent.tsx:2]: Fix the year

File: b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/profileSection/otherComponents/userDetails.tsx [hunk 12]
```diff
@@ -0,0 +1,114 @@
+/**
```
Comment (thivi) [inline b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/profileSection/otherComponents/userDetails.tsx:1]: Fix the year. Check the other places as well.

File: b2b-sample/apps/business-app/components/sections/storyHomes/guardioInsuranceHome.tsx [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+ * under the License.
+ */
+
+import { LogoComponent } from "@b2bsample/business-app/ui/ui-components";
+import { signout } from "@b2bsample/business-app/util/util-authorization-config-util";
+import { FooterComponent, HomeComponent, SignOutComponent } from "@b2bsample/shared/ui/ui-components";
+import { Session } from "next-auth";
+import { useState } from "react";
+import "rsuite/dist/rsuite.min.css";
+
```
Comment (thivi) [inline b2b-sample/apps/business-app/components/sections/storyHomes/guardioInsuranceHome.tsx:25]: Remove this empty line

File: b2b-sample/libs/business-admin-app/ui/ui-assets/src/lib/data/sideNav.json [hunk 14]
```diff
@@ -1,28 +1,70 @@
...
+                        "internal_idp_delete",
+                        "internal_application_mgt_view",
+                        "internal_application_mgt_update"
+                    ]
                 }
             ]
         }
     ]
-}
+}
```
Comment (thivi) [inline b2b-sample/libs/business-admin-app/ui/ui-assets/src/lib/data/sideNav.json:70]: Add a new line

**Review by thivi — APPROVED** [review]

File: b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/getStartedSectionComponent.tsx [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+        <FlexboxGrid align="middle" justify="space-between" style={ { height: "100%" } }>
+            <FlexboxGrid.Item colspan={ 14 }>
+                <GetStartedText />
+            </FlexboxGrid.Item>
+            <FlexboxGrid.Item colspan={ 9 }>
+                <Image src={ getStartedImage } alt="profile image" width={ 500 } />
+            </FlexboxGrid.Item>
+        </FlexboxGrid>
+    );
+
```
Comment (lashinijay) [inline b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/getStartedSectionComponent.tsx:40]: Shall we remove the unwanted new lines?

File: b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/otherComponents/getStartedText.tsx [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+                        <h4>Select one of the settings to get started</h4>
+                        <Button appearance="ghost" size="lg">
+                            <Stack spacing={ 3 } justifyContent="center" alignItems="center">
+                                <PagePreviousIcon />
+                                <p>Get Started</p>
+                            </Stack>
+                        </Button>
+                    </Stack>
+                </Stack>
+
```
Comment (lashinijay) [inline b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/otherComponents/getStartedText.tsx:57]: Do these new lines are required? check other places also.

File: b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/manageUserSection/otherComponents/editUserComponent.tsx [hunk 17]
```diff
@@ -130,7 +130,8 @@ export default function EditUserComponent(prop: EditUserComponentProps) {
             successTypeDialog(toaster, "Changes Saved Successfully", "User details edited successfully.");
             onClose();
         } else {
-            warningTypeDialog(toaster, "Roles not Updated", "Error occured while updating the roles. Try again.");
+            warningTypeDialog(toaster, "Roles not Properly Updated",
```
Comment (lashinijay) [inline b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/manageUserSection/otherComponents/editUserComponent.tsx:133]: `Roles not Properly Updated` -> `Roles are not properly updated`

File: b2b-sample/apps/business-admin-app/pages/o/[id].tsx [hunk 18]
```diff
@@ -17,16 +17,17 @@
...
+import { useEffect } from "react";
 import Home from "../../components/sections/home";
 
 export async function getServerSideProps(context) {
 
     const routerQuery = context.query.id;
     const session = await getSession(context);
 
-    if (session === null || session === undefined) {
+    if (session === null || session === undefined|| session.error) {
```
Comment (lashinijay) [inline b2b-sample/apps/business-admin-app/pages/o/[id].tsx:30]: `session === null || session === undefined|| session.error` -> `session === null || session === undefined || session.error`

File: b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/blogSection/otherComponents/newsComponent.tsx [hunk 19]
```diff
@@ -17,40 +17,44 @@
...
+            <Image src={ selectImage() } height={ 250 } width={ 300 } alt="news image" />
+            <Stack direction="column" alignItems="flex-start">
                 <p>{ getCurrentDate() }</p>
                 <br />
                 <h4>{ header }</h4>
                 <p>{ body }</p>
-            </Panel>
-        </div>
+            </Stack>
+
```
Comment (lashinijay) [inline b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/blogSection/otherComponents/newsComponent.tsx:51]: Let's remove the unwanted new line.

File: b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/phoneSection/otherComponents/phoneComponent.tsx [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+                                content="Looks like you don’t have Easy Meeting Phone enabled"
+                                headingType="h4"
+                                fontWeight="normal" />
+
+                            <Button appearance="primary">
+                                Upgrade to unlock this feature
+                            </Button>
+                        </Stack>
+
+
```
Comment (lashinijay) [inline b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/phoneSection/otherComponents/phoneComponent.tsx:42]: unnecessary new line

File: b2b-sample/apps/business-app/components/sections/storyHomes/easyMeetingsHome.tsx [hunk 21]
```diff
@@ -71,7 +89,7 @@ export default function Home(prop) {
 
                     <HomeComponent
                         scope={ session.scope }
-                        sideNavData = { sideNavData }
+                        sideNavData={ sideNavData }
                         activeKeySideNav={ activeKeySideNav }
```
Comment (lashinijay) [inline b2b-sample/apps/business-app/components/sections/storyHomes/easyMeetingsHome.tsx:93]: Shall we add spaces before and after = in other lines also to make them consistant

File: b2b-sample/apps/business-app/components/sections/storyHomes/guardioInsuranceHome.tsx [hunk 22]
```diff
@@ -0,0 +1,108 @@
...
+import sideNavGuardioInsuranceData
+    from "../../../../../libs/business-app/ui-assets/src/lib/data/sideNav-GuardioInsurance.json";
+import Custom500 from "../../../pages/500";
+import BlogSectionComponent from "../sections/GuardioInsurance/blogSection/blogSectionComponent";
+import ClaimsSectionComponent from "../sections/GuardioInsurance/claimsSection/claimsSectionComponent";
+import ProfileSectionComponent from "../sections/GuardioInsurance/profileSection/profileSectionComponent";
+import ReportsSectionComponent
+    from "../sections/GuardioInsurance/reportsSection/reportsSectionComponent";
+
+
```
Comment (lashinijay) [inline b2b-sample/apps/business-app/components/sections/storyHomes/guardioInsuranceHome.tsx:34]: unwanted new line

File: b2b-sample/libs/business-admin-app/data-access/data-access-controller/src/lib/controller/role/controllerEditRolesToAddOrRemoveUser/controllerDecodeEditRolesToAddOrRemoveUser.ts [hunk 23]
```diff
@@ -28,22 +29,35 @@ function getRolesThatNeedToRemoveUser(initRoleList: string[], roleList: string[]
...
 }
 
+async function patchRoleDetails(session: Session, userId: string, rolesUriList: string[], patchMethod: PatchMethod)
+    : Promise<boolean> {
+
+    for (const uri in rolesUriList) {
+        const role: Role | null = await controllerDecodePatchRole(session, uri, patchMethod, "users", [ userId ]);
+
+        if (!role) {
+
```
Comment (lashinijay) [inline b2b-sample/libs/business-admin-app/data-access/data-access-controller/src/lib/controller/role/controllerEditRolesToAddOrRemoveUser/controllerDecodeEditRolesToAddOrRemoveUser.ts:39]: unnecessary new lines. check other places also.

File: b2b-sample/libs/shared/util/util-front-end-util/src/lib/frontendUtil.tsx [hunk 24]
```diff
@@ -0,0 +1,130 @@
...
+ * @param itemScopes - scopes that required for this section
+ * 
+ * @returns `true` if user has the necessary scopes, else `false`
+ */
+function hideBasesdOnScopesSideNavItems(scopes: string[], itemScopes: string[]): boolean {
+    return itemScopes.every(scope => scopes.includes(scope));
+}
+
+/**
+ * hide content based on the user's realated privilages
```
Comment (lashinijay) [inline b2b-sample/libs/shared/util/util-front-end-util/src/lib/frontendUtil.tsx:49]: Start the sentence with a Capital letter and end with a full-stop.

File: b2b-sample/apps/business-app/components/sections/storyHomes/easyMeetingsHome.tsx (hunk 21 above)
Comment (Achintha444) [reply b2b-sample/apps/business-app/components/sections/storyHomes/easyMeetingsHome.tsx:93]: For props we do not use space before and after a `=`

File: b2b-sample/libs/business-admin-app/data-access/data-access-controller/src/lib/controller/role/controllerEditRolesToAddOrRemoveUser/controllerDecodeEditRolesToAddOrRemoveUser.ts (hunk 23 above)
Comment (Achintha444) [reply b2b-sample/libs/business-admin-app/data-access/data-access-controller/src/lib/controller/role/controllerEditRolesToAddOrRemoveUser/controllerDecodeEditRolesToAddOrRemoveUser.ts:39]: Space is required before a `return`.

File: b2b-sample/libs/shared/util/util-front-end-util/src/lib/frontendUtil.tsx (hunk 24 above)
Comment (Achintha444) [reply b2b-sample/libs/shared/util/util-front-end-util/src/lib/frontendUtil.tsx:49]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/8b059f39358c5652ea99e473f4cf34e23ad95e0b

File: b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/getStartedSectionComponent.tsx (hunk 15 above)
Comment (Achintha444) [reply b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/getStartedSectionComponent.tsx:40]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/c4229e64c82f57a16a579a24259eafab0e6cc56f

File: b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/otherComponents/getStartedText.tsx (hunk 16 above)
Comment (Achintha444) [reply b2b-sample/apps/business-admin-app/components/sections/sections/getStartedSection/otherComponents/getStartedText.tsx:57]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/c4229e64c82f57a16a579a24259eafab0e6cc56f

File: b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/manageUserSection/otherComponents/editUserComponent.tsx (hunk 17 above)
Comment (Achintha444) [reply b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/manageUserSection/otherComponents/editUserComponent.tsx:133]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/c4229e64c82f57a16a579a24259eafab0e6cc56f

File: b2b-sample/apps/business-admin-app/pages/o/[id].tsx (hunk 18 above)
Comment (Achintha444) [reply b2b-sample/apps/business-admin-app/pages/o/[id].tsx:30]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/c4229e64c82f57a16a579a24259eafab0e6cc56f

File: b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/phoneSection/otherComponents/phoneComponent.tsx (hunk 20 above)
Comment (Achintha444) [reply b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/phoneSection/otherComponents/phoneComponent.tsx:42]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/c4229e64c82f57a16a579a24259eafab0e6cc56f

File: b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/blogSection/otherComponents/newsComponent.tsx (hunk 19 above)
Comment (Achintha444) [reply b2b-sample/apps/business-app/components/sections/sections/EasyMeetings/blogSection/otherComponents/newsComponent.tsx:51]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/c4229e64c82f57a16a579a24259eafab0e6cc56f

File: b2b-sample/apps/business-app/components/sections/storyHomes/guardioInsuranceHome.tsx (hunk 22 above)
Comment (Achintha444) [reply b2b-sample/apps/business-app/components/sections/storyHomes/guardioInsuranceHome.tsx:34]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/c4229e64c82f57a16a579a24259eafab0e6cc56f

**Pull request merged**